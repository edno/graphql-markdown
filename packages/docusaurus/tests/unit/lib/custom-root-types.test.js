const { GraphQLFileLoader } = require("@graphql-tools/graphql-file-loader");

const { loadSchema, getSchemaMap } = require("../../../src/lib/graphql");

const SCHEMA_FILE = require.resolve(
  "../../__data__/schema_with_custom_root_types.graphql",
);
describe("graphql", () => {
  describe("schema with custom rootTypes", () => {
    let schema;
    beforeAll(async () => {
      schema = await loadSchema(SCHEMA_FILE, {
        loaders: [new GraphQLFileLoader()],
        rootTypes: { query: "Root", subscription: "" },
      });
    });

    test("loadSchema returns schema with custom root types", () => {
      expect.hasAssertions();

      expect(schema.getQueryType().name).toBe("Root");
      expect(schema.getMutationType()).toBeUndefined();
      expect(schema.getSubscriptionType()).toBeUndefined();
    });

    test("getSchemaMap returns a filtered map of schema types", () => {
      expect.hasAssertions();

      const schemaTypeMap = getSchemaMap(schema);

      expect(JSON.stringify(schemaTypeMap, null, 2)).toMatchSnapshot();
    });
  });
});
