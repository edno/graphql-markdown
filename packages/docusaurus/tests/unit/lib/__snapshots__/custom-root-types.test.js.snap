// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`graphql schema with custom rootTypes getSchemaMap returns a filtered map of schema types 1`] = `
"{
  "queries": {
    "allSubscription": {
      "name": "allSubscription",
      "type": "[Subscription]",
      "args": [
        {
          "name": "after",
          "type": "String",
          "extensions": {},
          "astNode": {
            "kind": "InputValueDefinition",
            "name": {
              "kind": "Name",
              "value": "after",
              "loc": {
                "start": 86,
                "end": 91
              }
            },
            "type": {
              "kind": "NamedType",
              "name": {
                "kind": "Name",
                "value": "String",
                "loc": {
                  "start": 93,
                  "end": 99
                }
              },
              "loc": {
                "start": 93,
                "end": 99
              }
            },
            "directives": [],
            "loc": {
              "start": 86,
              "end": 99
            }
          }
        },
        {
          "name": "first",
          "type": "Int",
          "extensions": {},
          "astNode": {
            "kind": "InputValueDefinition",
            "name": {
              "kind": "Name",
              "value": "first",
              "loc": {
                "start": 101,
                "end": 106
              }
            },
            "type": {
              "kind": "NamedType",
              "name": {
                "kind": "Name",
                "value": "Int",
                "loc": {
                  "start": 108,
                  "end": 111
                }
              },
              "loc": {
                "start": 108,
                "end": 111
              }
            },
            "directives": [],
            "loc": {
              "start": 101,
              "end": 111
            }
          }
        },
        {
          "name": "before",
          "type": "String",
          "extensions": {},
          "astNode": {
            "kind": "InputValueDefinition",
            "name": {
              "kind": "Name",
              "value": "before",
              "loc": {
                "start": 113,
                "end": 119
              }
            },
            "type": {
              "kind": "NamedType",
              "name": {
                "kind": "Name",
                "value": "String",
                "loc": {
                  "start": 121,
                  "end": 127
                }
              },
              "loc": {
                "start": 121,
                "end": 127
              }
            },
            "directives": [],
            "loc": {
              "start": 113,
              "end": 127
            }
          }
        },
        {
          "name": "last",
          "type": "Int",
          "extensions": {},
          "astNode": {
            "kind": "InputValueDefinition",
            "name": {
              "kind": "Name",
              "value": "last",
              "loc": {
                "start": 129,
                "end": 133
              }
            },
            "type": {
              "kind": "NamedType",
              "name": {
                "kind": "Name",
                "value": "Int",
                "loc": {
                  "start": 135,
                  "end": 138
                }
              },
              "loc": {
                "start": 135,
                "end": 138
              }
            },
            "directives": [],
            "loc": {
              "start": 129,
              "end": 138
            }
          }
        }
      ],
      "extensions": {},
      "astNode": {
        "kind": "FieldDefinition",
        "name": {
          "kind": "Name",
          "value": "allSubscription",
          "loc": {
            "start": 70,
            "end": 85
          }
        },
        "arguments": [
          {
            "kind": "InputValueDefinition",
            "name": {
              "kind": "Name",
              "value": "after",
              "loc": {
                "start": 86,
                "end": 91
              }
            },
            "type": {
              "kind": "NamedType",
              "name": {
                "kind": "Name",
                "value": "String",
                "loc": {
                  "start": 93,
                  "end": 99
                }
              },
              "loc": {
                "start": 93,
                "end": 99
              }
            },
            "directives": [],
            "loc": {
              "start": 86,
              "end": 99
            }
          },
          {
            "kind": "InputValueDefinition",
            "name": {
              "kind": "Name",
              "value": "first",
              "loc": {
                "start": 101,
                "end": 106
              }
            },
            "type": {
              "kind": "NamedType",
              "name": {
                "kind": "Name",
                "value": "Int",
                "loc": {
                  "start": 108,
                  "end": 111
                }
              },
              "loc": {
                "start": 108,
                "end": 111
              }
            },
            "directives": [],
            "loc": {
              "start": 101,
              "end": 111
            }
          },
          {
            "kind": "InputValueDefinition",
            "name": {
              "kind": "Name",
              "value": "before",
              "loc": {
                "start": 113,
                "end": 119
              }
            },
            "type": {
              "kind": "NamedType",
              "name": {
                "kind": "Name",
                "value": "String",
                "loc": {
                  "start": 121,
                  "end": 127
                }
              },
              "loc": {
                "start": 121,
                "end": 127
              }
            },
            "directives": [],
            "loc": {
              "start": 113,
              "end": 127
            }
          },
          {
            "kind": "InputValueDefinition",
            "name": {
              "kind": "Name",
              "value": "last",
              "loc": {
                "start": 129,
                "end": 133
              }
            },
            "type": {
              "kind": "NamedType",
              "name": {
                "kind": "Name",
                "value": "Int",
                "loc": {
                  "start": 135,
                  "end": 138
                }
              },
              "loc": {
                "start": 135,
                "end": 138
              }
            },
            "directives": [],
            "loc": {
              "start": 129,
              "end": 138
            }
          }
        ],
        "type": {
          "kind": "ListType",
          "type": {
            "kind": "NamedType",
            "name": {
              "kind": "Name",
              "value": "Subscription",
              "loc": {
                "start": 142,
                "end": 154
              }
            },
            "loc": {
              "start": 142,
              "end": 154
            }
          },
          "loc": {
            "start": 141,
            "end": 155
          }
        },
        "directives": [],
        "loc": {
          "start": 70,
          "end": 155
        }
      }
    },
    "subscription": {
      "name": "subscription",
      "type": "Subscription",
      "args": [
        {
          "name": "id",
          "type": "ID",
          "extensions": {},
          "astNode": {
            "kind": "InputValueDefinition",
            "name": {
              "kind": "Name",
              "value": "id",
              "loc": {
                "start": 171,
                "end": 173
              }
            },
            "type": {
              "kind": "NamedType",
              "name": {
                "kind": "Name",
                "value": "ID",
                "loc": {
                  "start": 175,
                  "end": 177
                }
              },
              "loc": {
                "start": 175,
                "end": 177
              }
            },
            "directives": [],
            "loc": {
              "start": 171,
              "end": 177
            }
          }
        }
      ],
      "extensions": {},
      "astNode": {
        "kind": "FieldDefinition",
        "name": {
          "kind": "Name",
          "value": "subscription",
          "loc": {
            "start": 158,
            "end": 170
          }
        },
        "arguments": [
          {
            "kind": "InputValueDefinition",
            "name": {
              "kind": "Name",
              "value": "id",
              "loc": {
                "start": 171,
                "end": 173
              }
            },
            "type": {
              "kind": "NamedType",
              "name": {
                "kind": "Name",
                "value": "ID",
                "loc": {
                  "start": 175,
                  "end": 177
                }
              },
              "loc": {
                "start": 175,
                "end": 177
              }
            },
            "directives": [],
            "loc": {
              "start": 171,
              "end": 177
            }
          }
        ],
        "type": {
          "kind": "NamedType",
          "name": {
            "kind": "Name",
            "value": "Subscription",
            "loc": {
              "start": 180,
              "end": 192
            }
          },
          "loc": {
            "start": 180,
            "end": 192
          }
        },
        "directives": [],
        "loc": {
          "start": 158,
          "end": 192
        }
      }
    }
  },
  "directives": [
    "@include",
    "@skip",
    "@deprecated",
    "@specifiedBy"
  ],
  "objects": {
    "Subscription": "Subscription"
  },
  "unions": {},
  "interfaces": {},
  "enums": {},
  "inputs": {},
  "scalars": {
    "String": "String",
    "ID": "ID",
    "Int": "Int",
    "Boolean": "Boolean"
  }
}"
`;
